cpp/src/slice2java/JavaUtil.h

```diff
@@ -8,106 +8,72 @@
-    //
-    std::string getSerialVersionUID(const ContainedPtr&);
-    std::int64_t computeDefaultSerialVersionUID(const ContainedPtr&);
-
-    //
-    // Returns true if we can generate a method from the given data member list. A Java method
-    // can have a maximum of 255 parameters (including the implicit 'this') where each parameter
-    // is counted as 1 unit, except for long and double which are counted as 2 units.
-    // See https://docs.oracle.com/javase/specs/jvms/se20/html/jvms-4.html#jvms-4.3.3
-    //
-    bool isValidMethodParameterList(const DataMemberList&, int additionalUnits = 0);
-
-    /// Returns true if and only if 'p' maps to one of the builtin Java types (ie. a primitive type or a string).
-    bool mapsToJavaBuiltinType(const TypePtr& p);
+        std::string getResultType(const OperationPtr&, const std::string&, bool, bool);
 
-    class JavaOutput final : public ::IceInternal::Output
-    {
-    public:
-        JavaOutput();
+        /// Returns a vector of this operation's parameters with each of them formatted as 'paramType paramName'.
+        /// If 'internal' is true, the names will be prefixed with "iceP_".
+        std::vector<std::string>
+        getParamsProxy(const OperationPtr& op, const std::string& package, bool optionalMapping, bool internal = false);
+
+        /// Returns a vector of this operation's parameter's names in order.
+        /// If 'internal' is true, the names will be prefixed with "iceP_".
+        std::vector<std::string> getInArgs(const OperationPtr& op, bool internal = false);
 
         //
-        // Open a file to hold the source for a Java class. The first
-        // argument is the class name (including an optional leading
-        // package). Intermediate directories will be created as
-        // necessary to open the file in the package. The second
-        // argument specifies a directory prefix in which to locate
-        // the class.
-        //
-        // After successfully opening the file, the function invokes
-        // printHeader() and then emits a "package" statement if
-        // necessary.
+        // These functions should only be called for classes, exceptions, and structs.
+        // Enums automatically implement Serializable (Java just serializes the enumerator's identifier),
+        // and proxies get their implementation from `_ObjectPrxI`.
         //
-        void openClass(const std::string&, const std::string&, const std::string& = std::string());
+        std::string getSerialVersionUID(const ContainedPtr&);
+        std::int64_t computeDefaultSerialVersionUID(const ContainedPtr&);
 
-        void printHeader();
-    };
-
-    class JavaGenerator
-    {
-    public:
-        JavaGenerator(const JavaGenerator&) = delete;
-        virtual ~JavaGenerator();
-
-        JavaGenerator& operator=(const JavaGenerator&) = delete;
-
-        static void validateMetadata(const UnitPtr&);
-
-        void close();
-
-        JavaGenerator(std::string);
```

Comment:
Most of the methods in this class were static, and this PR moves them outside the class, converting them into free functions.
Git diff interprets this as if I cut-and-pasted this section at the bottom of the file.